### ISSUE TITLE
zone.js drains microtasks too early in synchronous event call

### ISSUE THREAD

**Issue by xiaody** [angular/angular#44446]
### Which @angular/* package(s) are the source of the bug?

Don't known / other

### Is this a regression?

No

### Description

Please see the reproduction. It seems zone.js assumes EventTarget's event call is always a standalone task, which may not be true.

```js
const invoke = (fn) => {
  document.body.addEventListener('click', fn);
  document.body.click();
  document.body.removeEventListener('click', fn);
};

invoke(() => {
  Promise.resolve().then(() => console.log(2));
});
console.log(1);
```

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/typescript-88btqj?file=index.ts

### Please provide the exception or error you saw

```true
Native Chrome/Safari/Firefox outputs `1 2`, while with zone.js outputs `2 1`.
```

### Anything else?

I noticed this during debugging https://github.com/angular/angular/issues/41506#issuecomment-991765487 , maybe related.

**Comment by JiaLiPassion** [comment]
@xiaody , yes, this is a known issue, and may not be fixed in the near future, it there any real world application has some issues because this behavior?

**Comment by xiaody** [comment]
React's `invokeGuardedCallbackImpl` [uses such pattern](https://github.com/facebook/react/blob/4e6eec69be632c0c0177c5b1c8a70397d92ee181/packages/shared/invokeGuardedCallbackImpl.js#L190-L195), it creates a `fakeNode`, add a listener on it and dispatch an event synchronously. #44446(with old version antd) maybe a real world case as I have commented there.
